nautobot/extras/views.py

```diff
@@ -1567,6 +1570,51 @@ def get(self, request, pk=None):
         return HttpResponse(log_table.as_html(request))
 
 
+#
+# Job Button
+#
+
+
+class JobButtonUIViewSet(NautobotUIViewSet):
+    filterset_class = filters.JobButtonFilterSet
+    filterset_form_class = forms.JobButtonFilterForm
+    form_class = forms.JobButtonForm
+    lookup_field = "pk"
+    queryset = JobButton.objects.all()
+    serializer_class = serializers.JobButtonSerializer
+    table_class = tables.JobButtonTable
+
+
+class JobButtonRunView(ObjectPermissionRequiredMixin, View):
+    """
+    View to run the Job linked to the Job Button.
+    """
+
+    queryset = JobButton.objects.all()
+
+    def get_required_permission(self):
+        return "extras.run_job"
+
+    def post(self, request, pk):
+        post_data = request.POST
+        job_button = JobButton.objects.get(pk=pk)
+        job_model = job_button.job
+        result = JobResult.enqueue_job(
+            func=run_job,
+            name=job_model.class_path,
+            obj_type=get_job_content_type(),
+            user=request.user,
+            data={
+                "obj": post_data["obj"],
+            },
+            request=copy_safe_request(request),
+            commit=True,
+        )
```

Comment:
Can we make the signature for this more like `JobHookReceiver`? The current implementation appears it will allow a user to associate a Job that doesn't take the object on the `obj` key.

https://github.com/nautobot/nautobot/blob/develop/examples/example_plugin/example_plugin/jobs.py#L60-L81

Also, I'm assuming if I read this correctly the only jobs candidates for Job Buttons do not allow other arguments to be entered into them?

Reply:
> Can we make the signature for this more like JobHookReceiver? The current implementation appears it will allow a user to associate a Job that doesn't take the object on the obj key.

I'll look at it and see the difference would be.

> Also, I'm assuming if I read this correctly the only jobs candidates for Job Buttons do not allow other arguments to be entered into them?

Yes, this is [based on the discussion](https://github.com/nautobot/nautobot/issues/3033#issuecomment-1380968777) on the linked issue:

> The job would have to have no input.